EventFilter/Utilities/plugins/GlobalEvFOutputModule.cc

```diff
@@ -429,7 +427,16 @@ namespace evf {
       edm::LuminosityBlockForOutput const& iLB) const {
     auto openDatFilePath = edm::Service<evf::EvFDaqDirector>()->getOpenDatFilePath(iLB.luminosityBlock(), streamLabel_);
 
-    return std::make_shared<GlobalEvFOutputEventWriter>(openDatFilePath, iLB.luminosityBlock());
+    auto ret = std::make_shared<GlobalEvFOutputEventWriter>(openDatFilePath, iLB.luminosityBlock());
+
+    StreamerOutputModuleCommon streamerCommon(
+        commonParameters_, &keptProducts()[edm::InEvent], description().moduleLabel());
+
+    auto msg = streamerCommon.serializeEventMetaData(
+        *streamerCommon.getSerializerBuffer(), *branchIDLists(), *thinnedAssociationsHelper());
+
+    ret->doOutputEvent(*msg);
+    return ret;
```

Comment:
A comment on possible optimization (maybe not this PR):
As this is repeated every lumi block (and will be done 50-70 times per process, for each stream in HLT), would it be feasible to serializ the buffer and cache it in beginRun? Then it only needs to be written out without discarding each time (I think msg is a shared_ptr, so it's fine if it's cached).
Writing could also be postponed to the first event being written (in the async block), because sometimes there is no event in a stream in a lumi in HLT (not expecting a big impact, since this is about ~ 60 kB uncompressed).

Reply:
event meta data changes are now cached.